nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1625,6 +1625,17 @@ public DocumentModel saveDocument(DocumentModel docModel) {
 
         notifyEvent(DocumentEventTypes.DOCUMENT_UPDATED, docModel, options, null, null, true, false);
         updateDocumentCount.inc();
+
+        // Notify that proxies have been updated
+        List<Document> proxies = getSession().getProxies(doc, null);
```

Comment:
`getProxies(doc, null)` is costly because it always does a search. It would be better to add a new `getProxies(doc)` API that is simpler and ends up using `KEY_PROXY_IDS` for DBS, and `PersistenceContext.targetProxies` for VCS.